library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -462,7 +462,7 @@ impl RwLock {
                 // not been completed yet, so we still have exclusive access.
                 // Retract the downgrade request and unlock, but leave waking up new threads to the
                 // thread that already holds the queue lock.
-                Some(state.wrapping_byte_sub(DOWNGRADED | LOCKED))
+                Some(state.map_addr(|addr| addr & !(DOWNGRADED | LOCKED)))
```

Comment:
When I made my suggestion I forgot that `some_ptr.mask(MASK)` exists, and is actually (iiuc) better than `some_ptr.map_addr(|addr| addr & MASK)`. So these should probably be changed where `&` is used (there is no equivalent for `|`).

You can probably squash the last two commits then (6db6e878d6a51e6e578369a762416aa3d63ccce4 and 1da26a301a6a7d99d44a1ce890b0de6a1e766d01) since "change wrapping_byte_*s to map_addrs " is just a fixup.